### PR TITLE
Newsletter in the footer if cms block is visible

### LINKED ISSUE #1398: Newsletter in the footer / M2 Support
Magento 2 has a newsletter subscription in the footer field by default. We need to implement support for this feature. 

DESIGN
https://prnt.sc/uvgu4q

TODO
* Add newsletter subscription field to the footer 
* Style it as per design on both mobile and desktop
* Add placeholder text "Enter your email address"
** The text should be translatable 
* The orange button in the design indicates the submit button 
** On click on the submit button, the customer should be subscribed to the newsletter 
* Add email validation to the subscription field (same as during checkout)
* On submit button:
** trigger email validation
*** if email failed validation, reuse email field behavior such as we have in checkout ( change bolder color to red and show “Email is invalid.” https://prnt.sc/sx9zq7)
*** show success notification if the email was added to the newsletter subscription. Use the following text: “You’ve successfully subscribed to our newsletter”
*** The text should be translatable
*** If validation is correct, the email address is added to the newsletter subscribers list

EXPECTED RESULT
* Newsletter subscription in the footer is styled 
* Email validation is present
* Success/error notifications are shown
* Customers can be subscribed to newsletter

### PR DESCRIPTION
**Pull request by zans-laksa**
Fixes #1398
Added ability to display newsletter subcription component if footer contents comes from the cms.

### CODE REVIEW

File: src/app/component/Footer/Footer.component.js [hunk 1]
```diff
@@ -121,11 +127,33 @@ export class Footer extends PureComponent {
...
+            </div>
+        );
+    }
+
     renderContent() {
         const { footer_content: { footer_cms } = {} } = window.contentConfiguration;
 
         if (footer_cms) {
-            return <CmsBlock identifier={ footer_cms } />;
+            return this.renderCmsBlockWrapper(footer_cms);
```
Comment (alfredsgenkins) [inline src/app/component/Footer/Footer.component.js:156]: No need to pass this `footer_cms` as an argument, take it from props in the funct

File: src/app/component/Footer/Footer.style.scss [hunk 2]
```diff
@@ -133,4 +133,20 @@
...
+            flex: 1;
+        }
+
+        .ContentWrapper .Footer-Columns {
+            padding: 0;
+        }
+
+        .Footer-Column_isNewsletter {
+            @include desktop {
+                padding-left: 100px;
```
Comment (alfredsgenkins) [inline src/app/component/Footer/Footer.style.scss:148]: This is strange padding, does it work on small screens?

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]

File: src/app/component/Footer/Footer.component.js (hunk 1 above)
Comment (zans-laksa) [reply src/app/component/Footer/Footer.component.js]: Done

File: src/app/component/Footer/Footer.style.scss (hunk 2 above)
Comment (zans-laksa) [reply src/app/component/Footer/Footer.style.scss:148]: On small screens the block content's layout is vertical and centered.

**Review by yeegor — APPROVED** [review]
Leģit

**Pull request merged**